src/net/gosha/atproto/client.clj

```diff
@@ -20,6 +22,17 @@
    :username "ATPROTO_USERNAME"
    :app-password "ATPROTO_APP_PASSWORD"})
 
+(def ^{:private true
+       :doc "access and refresh tokens for the API"}
+  tokens
+  (atom {:access nil :refresh nil}))
```

Comment:
I think we want the tokens to be an atom, as discussed, so they can be seamlessly replaced, without needing to return a (possibly updated session) from every API call. But that doesn't mean they should be a _global_ atom. Having them global means this code would never be able to manage two different sessions in the same process.

Instead, the atom itself should be associated with the session, so if we have different sessions we can have different atoms.

So how do we do that, given that our session is an instance of `martian.core.Martian`?

Well, `Martian` is a record, which means it's already an arbitrary map. Notice on how line 122 we're storing our own config in the session map for later reference: `(assoc session ::config config)`. We use a namespaced key to make sure there's never any collisions with something Martian is doing. 

My recommendation would be to do the same thing with the tokens: `(assoc session ::tokens (atom {:access access :refresh refresh}))`.

Then they're available whenever we need them, can be updated statefully, and are still part of the session instead of globals.